input-profile.js

```diff
@@ -0,0 +1,249 @@
+import {Component, Property} from '@wonderlandengine/api';
+import {vec3, quat} from 'gl-matrix';
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    static Properties = {
+        controller: Property.object(),
+        handednessIndex: Property.enum(['left', 'right'], 0),
+        path: Property.string(
+            'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@1.0.13/dist/profiles/'
+        ),
+        customProfileFolder: Property.string(''),
+        preSpawnedController: Property.object(),
+    };
+
+    init() {
+        this._gamepadObjects = {};
```

Comment:
A more efficient data structure, like an index-base mapping computed from the original input profile would allow to loop through each gamepadobject and update them one by one